proxy_agent/src/redirector.rs

```diff
@@ -45,14 +45,14 @@ impl AuditEntry {
 
 const MAX_STATUS_MESSAGE_LENGTH: usize = 1024;
 
-pub fn start_async(local_port: u16, shared_state: Arc<Mutex<SharedState>>) {
-    thread::spawn(move || {
-        start(local_port, shared_state);
+pub async fn start_async(local_port: u16, shared_state: Arc<Mutex<SharedState>>) {
+    tokio::spawn(async move {
+        start(local_port, shared_state).await;
     });
```

Comment:
This function doesn't need to be marked `async` since it never needs to await anything. Additionally, there's no need to wrap the `start()` function in an async block as it's already returning a future you can pass to `tokio::spawn`:
```suggestion
pub fn start_async(local_port: u16, shared_state: Arc<Mutex<SharedState>>) {
    tokio::spawn(start(local_port, shared_state));
```

This applies to a number of other functions in this PR as well.

Reply:
removed the "async move" within all tokio::spwan function.